neuraxle/steps/numpy.py

```diff
@@ -373,3 +374,198 @@ def __init__(self, new_shape):
+        :class:`~neuraxle.base.NonFittableMixin`
+    """
+    def __init__(self, axis=None):
+        BaseStep.__init__(self)
+        NonFittableMixin.__init__(self)
+
+        if axis is None:
+            axis = -2
+        self.axis = axis
+
+    def transform(self, data_inputs):
+        """
+        Will featurize data with a median.
+
+        :param data_inputs: data inputs
+        :return: data inputs median for the given axis
+        """
+        return np.median(data_inputs, axis=self.axis)
+
+
+class NumpyMin(NonFittableMixin, BaseStep):
+    """
+    Compute `np.min <https://numpy.org/doc/1.18/reference/generated/numpy.minimum.html>`_ at the given axis.
+
+    .. seealso::
+        :class:`~neuraxle.base.BaseStep`,
+        :class:`~neuraxle.base.NonFittableMixin`
+    """
+    def __init__(self, axis=None):
+        BaseStep.__init__(self)
+        NonFittableMixin.__init__(self)
+
+        if axis is None:
+            axis = -2
+        self.axis = axis
+
+    def transform(self, data_inputs):
+        """
+        Will featurize data with a min.
+
+        :param data_inputs: data inputs
+        :return: min value for the given axis
+        """
+        return np.min(data_inputs, axis=self.axis)
+
+
+class NumpyMax(NonFittableMixin, BaseStep):
+    """
+    Compute `np.max <https://numpy.org/doc/1.18/reference/generated/numpy.ndarray.max.html>̀_ at the given axis.
+
+    .. seealso::
+        :class:`~neuraxle.base.BaseStep`,
+        :class:`~neuraxle.base.NonFittableMixin`
+    """
+    def __init__(self, axis=None):
+        BaseStep.__init__(self)
+        NonFittableMixin.__init__(self)
+
+        if axis is None:
+            axis = -2
+        self.axis = axis
+
+    def transform(self, data_inputs):
+        """
+        Will featurize data with a max.
+
+        :param data_inputs: 3D time series of shape [batch_size, time_steps, sensors]
+        :return: max value for the given axis
+        """
+        return np.max(data_inputs, axis=self.axis)
+
+
+class FFTPeakBinWithValue(NonFittableMixin, BaseStep):
+    """
+    Compute peak fft bins (int), and their magnitudes' value (float), to concatenate them.
+
+    .. seealso::
+        :class:`~neuraxle.base.BaseStep`,
+        :class:`~neuraxle.base.NonFittableMixin`
+    """
+    def transform(self, data_inputs):
+        """
+        Will compute peak fft bins (int), and their magnitudes' value (float), to concatenate them.
+
+        :param data_inputs: real magnitudes of an fft. It could be of shape [batch_size, bins, features].
+        :return: Two arrays without bins concatenated on feature axis. Shape: [batch_size, 2 * features]
+        """
+        time_bins_axis = -2
+        peak_bin = np.argmax(data_inputs, axis=time_bins_axis)
+        peak_bin_val = np.max(data_inputs, axis=time_bins_axis)
+
+        # Notice that here another FeatureUnion could have been used with a joiner:
+        transformed = np.concatenate([peak_bin, peak_bin_val], axis=-1)
```

Comment:
```suggestion
        # Notice that here another `FeatureUnion` or `ChooseOneOrManyStepsOf` could have been used with a joiner:
        transformed = np.concatenate([peak_bin, peak_bin_val], axis=-1)
```

Reply:
For extremists. This could be resolved at the same time of the other comment above.